### PR TITLE
fix: do not use /etc/sysconfig/kdump, use /etc/kdump.conf for everything

### PR DESCRIPTION
**Pull request by licliu**
Currently kexec-tools will generate /etc/sysconfig/kdump when installed.
We no longer need to manage this file, these tasks are no longer necessary.

Add support for auto_reset_crashkernel and dracut_args in kdump
configuration file.

kdumpctl reset-crashkernel only supports RHEL 9 and later versions,
use grubby to update crashkernel for versions before RHEL 9.

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -30,19 +30,6 @@
...
-    path: /etc/sysconfig/kdump
-  register: __sysconfig_kdump
-
-- name: Generate /etc/sysconfig/kdump
-  template:
-    src: kdump.j2
-    dest: /etc/sysconfig/kdump
-    backup: true
-    mode: "{{ __sysconfig_kdump.stat.mode | d('0644') }}"
-  notify: Restart kdump
```
Comment (richm) [inline tasks/main.yml:44]: Do we still need this for rhel6?  rhel7?  If so, we can put a conditional here.

File: templates/kdump.conf.j2 [hunk 2]
```diff
@@ -18,3 +18,17 @@ path {{ kdump_path }}
 core_collector {{ kdump_core_collector }}
 {% endif %}
 default {{ kdump_system_action }}
+
+{% if ansible_facts['distribution'] in ['RedHat', 'CentOS'] %}
```
Comment (richm) [inline templates/kdump.conf.j2:22]: What about Fedora?  If this applies to Fedora too, then you can use
```suggestion
{% if ansible_facts['os_family'] == 'Fedora' %}
```

File: tasks/main.yml (hunk 1 above)
Comment (licliu) [reply tasks/main.yml:44]: Oh, I haven't touched RHEL 6 for a long time, but fortunately, both RHEL6 and RHEL 7 versions of kexec-tools can generate this file.

File: templates/kdump.conf.j2 [hunk 3]
```diff
@@ -18,3 +18,17 @@ path {{ kdump_path }}
...
+{% if ansible_facts['distribution_major_version'] | int >= 9 %}
+{% if kdump_auto_reset_crashkernel %}
+auto_reset_crashkernel yes
+{% else %}
+auto_reset_crashkernel no
+{% endif %}
+{% endif %}
+{% endif %}
+
+{% if kdump_dracut_args %}
```
[the remark this answers is not in the document]
Comment (licliu) [reply templates/kdump.conf.j2:32]: This option is supported from RHEL 7, sorry forgot to consider RHEL 6.

File: templates/kdump.conf.j2 (hunk 2 above)
Comment (licliu) [reply templates/kdump.conf.j2:22]: Fedora supports it starting from f35.

File: tasks/main.yml [hunk 4]
```diff
@@ -61,6 +48,13 @@
     - ansible_facts['distribution'] in ['RedHat', 'CentOS']
     - ansible_facts['distribution_major_version'] | int >= 9
 
+- name: Use grubby to update crashkernel=auto if needed
+  ansible.builtin.command: grubby --args="crashkernel=auto" --update-kernel=ALL
+  when:
+    - kdump_reboot_required | bool
+    - ansible_facts['distribution'] in ['RedHat', 'CentOS']
```
[the remark this answers is not in the document]
Comment (licliu) [reply tasks/main.yml:55]: I have searched about os_family, seems that "RedHat" will catch the following distributions:
```
RedHat
CentOS
Fedora
Scientific
CloudLinux
OracleLinux
Amazon
XenServer
```

I'm not sure if all these distributions use the same majro version number, so I just use "distribution".

File: tasks/main.yml [hunk 5]
```diff
@@ -61,6 +48,13 @@
     - ansible_facts['distribution'] in ['RedHat', 'CentOS']
     - ansible_facts['distribution_major_version'] | int >= 9
 
+- name: Use grubby to update crashkernel=auto if needed
+  ansible.builtin.command: grubby --args="crashkernel=auto" --update-kernel=ALL
```
[the remark this answers is not in the document]
Comment (licliu) [reply tasks/main.yml:52]: OK, got it.

File: tasks/main.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (richm) [reply tasks/main.yml:55]: ok - the issue is we keep getting requests for other system roles to support clones like Alma, Rocky, Oracle, etc. which do use the same version numbering - and it gets very cumbersome to use `ansible_facts['distribution'] in ['RedHat', 'CentOS', 'AlmaLinux', 'Rocky', 'Oracle']` and having to keep updating this list - but I guess in this case if the major version does matter, we'll have to use `['RedHat', 'CentOS']`, and wait until someone asks us for support for Alma, etc.

File: templates/kdump.conf.j2 (hunk 2 above)
Comment (richm) [reply templates/kdump.conf.j2:22]: ok - that's fine - we only support the current versions of Fedora

**Comment by richm** [comment]
[citest]

File: defaults/main.yml [hunk 6]
```diff
@@ -7,3 +7,5 @@ kdump_ssh_user: null
 kdump_ssh_server: null
 kdump_sshkey: /root/.ssh/kdump_id_rsa
 kdump_reboot_ok: false
+kdump_auto_reset_crashkernel: true
+kdump_dracut_args: null
```
[the remark this answers is not in the document]
Comment (licliu) [reply defaults/main.yml]: I got a lint error after adding the newline, strange.

**Comment by spetrosi** [comment]
[citest]

File: defaults/main.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (richm) [reply defaults/main.yml]: yes - now getting an error about too many newlines - try removing one of them

**Comment by richm** [comment]
Getting an error on rhel6:
```
TASK [linux-system-roles.kdump : Ensure that kdump is started] *****************
task path: /WORKDIR/dist-git-kdump-lichliu-bu98a16c/tests/roles/linux-system-roles.kdump/tasks/main.yml:83
Monday 03 April 2023  09:19:35 +0000 (0:00:00.122)       0:00:07.375 ********** 
fatal: [sut]: FAILED! => {
    "changed": false
}

MSG:

Kdump is not supported on this kernel[FAILED]
```

**Comment by licliu** [comment]
> Getting an error on rhel6:
> 
> ```
> TASK [linux-system-roles.kdump : Ensure that kdump is started] *****************
> task path: /WORKDIR/dist-git-kdump-lichliu-bu98a16c/tests/roles/linux-system-roles.kdump/tasks/main.yml:83
> Monday 03 April 2023  09:19:35 +0000 (0:00:00.122)       0:00:07.375 ********** 
> fatal: [sut]: FAILED! => {
>     "changed": false
> }
> 
> MSG:
> 
> Kdump is not supported on this kernel[FAILED]
> ```

I spent some time to check why this PR has this error, but I can't reproduce it on my local centos 6 vm.
I checked some previous PRs and found that some PRs containing [citest] also have this error. 
Can you help me to check if there is a problem with the image used for testing?

**Comment by richm** [comment]
@ukulekek any idea about the rhel 6 images?  Local rhel6 and centos6 qemu qcow VMs work fine.

**Comment by licliu** [comment]
Hello, I will be on vacation next, and I will be back on May 3rd, thank you for your understanding.

**Comment by licliu** [comment]
@richm @spetrosi Hey there! I'm just checking in on this PR to see if you've had a chance to review yet. Let me know if you have any questions or need any changes from me. Thanks!

File: templates/kdump.conf.j2 (hunk 3 above)
[the remark this answers is not in the document]
Comment (spetrosi) [reply templates/kdump.conf.j2:32]: So will this work on RHEL6?

**Comment by spetrosi** [comment]
[citest]

File: templates/kdump.conf.j2 (hunk 3 above)
[the remark this answers is not in the document]
Comment (licliu) [reply templates/kdump.conf.j2:32]: I have fixed that last month, local rhel6 and centos6 qemu qcow VMs work fine, but the ci test failed with "Kdump is not supported on this kernel[FAILED]".

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
What happens if `/etc/sysconfig/kdump` is still on the system after running this role?  Should we remove that file?

Also - since you added this PR we have switched to using Conventional Commits format for commit and PR comments - see https://linux-system-roles.github.io/contribute.html "Write a good commit message" e.g. something like

```
fix: do not use /etc/sysconfig/kdump.conf, use /etc/kdump.conf for everything

do not use /etc/sysconfig/kdump.conf, use /etc/kdump.conf for everything, including
crashkernel and dracut configuration
use grubby to configure kernel `crashkernel=auto` on systems that need it upon reboot
```
or whatever

**Comment by licliu** [comment]
> What happens if `/etc/sysconfig/kdump` is still on the system after running this role? Should we remove that file?

Let's just leave it as it is.
 
> 
> Also - since you added this PR we have switched to using Conventional Commits format for commit and PR comments - see https://linux-system-roles.github.io/contribute.html "Write a good commit message" e.g. something like
> 
> ```
> fix: do not use /etc/sysconfig/kdump.conf, use /etc/kdump.conf for everything
> 
> do not use /etc/sysconfig/kdump.conf, use /etc/kdump.conf for everything, including
> crashkernel and dracut configuration
> use grubby to configure kernel `crashkernel=auto` on systems that need it upon reboot
> ```
> 
> or whatever

OK, I just rebase the new commits ,renamed the pr and changed some codes to make ansible-lint happy.

**Pull request merged**